src/EditorFeatures/Test/Workspaces/ClassificationTypeNamesTests.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Reflection;
+using Microsoft.CodeAnalysis.Classification;
+using Microsoft.VisualStudio.Text.Classification;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces
+{
+    public class ClassificationTypeNamesTests
+    {
+        public static IEnumerable<object[]> AllClassificationTypeNames
+        {
+            get
+            {
+                foreach (var field in typeof(ClassificationTypeNames).GetFields(BindingFlags.Static | BindingFlags.Public))
+                {
+                    yield return new object[] { field.Name, field.GetRawConstantValue() };
+                }
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(AllClassificationTypeNames))]
+        [WorkItem(25716, "https://github.com/dotnet/roslyn/issues/25716")]
+        public void ClassificationTypeExported(string fieldName, object constantValue)
+        {
+            Assert.IsType<string>(constantValue);
+            string classificationTypeName = (string)constantValue;
+            var exportProvider = TestExportProvider.ExportProviderWithCSharpAndVisualBasic;
+            var exports = exportProvider.GetExports<ClassificationTypeDefinition, IClassificationTypeDefinitionMetadata>();
```

Comment:
Should this just import the IContentTypeRegistryService and ask that for the content type by name? I can imagine cases where we could screw something up (cycles in base types, maybe?) that might pass this.

Reply:
I was thinking that type wouldn't be available in unit tests, but maybe it is. I'll switch to it if available.